docs/protocol/extension_statistics.md

```diff
@@ -42,72 +42,125 @@ $object and $array refer to the fundamental JSON types. #optional
 }
 ```
 
+- "name" : The name of the model.
+
+- "version" : The version of the model.
+
+- "last_inference" : The timestamp of the last inference request made
+  for this model, as milliseconds since the epoch.
+
+- "inference_stats" : The aggregate statistics for the
+  model/version. So, for example, "inference_stats":"success"
+  indicates the number of successful inference requests for the model.
+
+- "batch_stats" : The aggregate statistics for each different batch
+  size that is executed in the model. The batch statistics indicate
+  how many actual model executions were performed and show differences
+  due to different batch size (for example, larger batches typically
+  take longer to compute).
+
 ```
-$stats =
+$inference_stats =
 {
-  "inference" : {
-    "success" : $duration_stat,
-    "fail" : $duration_stat,
-    "queue" : $duration_stat,
-    "compute_input" : $duration_stat,
-    "compute_infer" : $duration_stat,
-    "compute_output" : $duration_stat
-  }
+  "success" : $duration_stat,
+  "fail" : $duration_stat,
+  "queue" : $duration_stat,
+  "compute_input" : $duration_stat,
+  "compute_infer" : $duration_stat,
+  "compute_output" : $duration_stat
 }
 ```
 
-- “inference:success” : The count and cumulative duration for all
-  successful inference requests.
-
-- “inference:fail” : The count and cumulative duration for all failed
+- “success” : The count and cumulative duration for all successful
   inference requests.
 
-- “inference:queue” : The count and cumulative duration that inference
-  requests wait in scheduling or other queues.
+- “fail” : The count and cumulative duration for all failed inference
+  requests.
+
+- “queue” : The count and cumulative duration that inference requests
+  wait in scheduling or other queues.
 
-- “inference:compute_input” : The count and cumulative duration to
-  prepare input tensor data as required by the model framework /
-  backend. For example, this duration should include the time to copy
-  input tensor data to the GPU.
+- “compute_input” : The count and cumulative duration to prepare input
+  tensor data as required by the model framework / backend. For
+  example, this duration should include the time to copy input tensor
+  data to the GPU.
 
-- “inference:compute_infer” : The count and cumulative duration to
-  execute the model.
+- “compute_infer” : The count and cumulative duration to execute the
+  model.
 
-- “inference:compute_output” : The count and cumulative duration to
-  extract output tensor data produced by the model framework /
-  backend. For example, this duration should include the time to copy
-  output tensor data from the GPU.
+- “compute_output” : The count and cumulative duration to extract
+  output tensor data produced by the model framework / backend. For
+  example, this duration should include the time to copy output tensor
+  data from the GPU.
+
+```
+$batch_stats =
+{
+  "batch_size" : $number,
+  "count" : $number,
```

Comment:
"count" can be obtained from `$duration_stat`, will it be redundant?

Reply:
@GuanLuo  Yes, good point. Can you remove it as part of your stats change.